pkg/controllers/resources/pods/translate/translator.go

```diff
@@ -414,7 +431,22 @@ func (t *translator) translateVolumes(pPod *corev1.Pod, vPod *corev1.Pod) error
 	return nil
 }
 
-func (t *translator) translateProjectedVolume(projectedVolume *corev1.ProjectedVolumeSource, pPod *corev1.Pod, vPod *corev1.Pod) error {
+func (t *translator) shouldCreateTokenSecret(vPod *corev1.Pod) bool {
+	// if the automount field is not set at all (which means true by default)
+	// or if set, then should be true, skip otherwise
+	if vPod.Spec.AutomountServiceAccountToken != nil &&
+		!*vPod.Spec.AutomountServiceAccountToken {
+		return false
+	}
```

Comment:
I am not 100% sure if this is a reliable way to check if a SA token is needed.
It would be best to check if there is a projected volume of `serviceAccountToken` type.
AFAIK, when `AutomountServiceAccountToken != false` the pod has a serviceAccountToken projected volume added in spec, but it might be added for a different reason too.